src/current/v23.2/changefeed-messages.md

```diff
@@ -191,70 +191,134 @@ CREATE CHANGEFEED FOR TABLE rides INTO 'external://kafka' WITH diff, envelope=wr
+    {"after": {"id": 4, "name": "Danny", "office": "los angeles"}, "key": [4], "updated": "1701102561022789676.0000000000"}
     ~~~
 
-- If a row is modified more than once in the same transaction, only the last change will be emitted.
+    The messages received at the sink are in order by timestamp **for each key**. Here, the update for key `[1]` has emitted before the insertion of key `[2]` even though the timestamp for the update to key `[1]` is higher. If you order messages by key, they will always be in the correct timestamp order.
 
-- Rows are sharded between Kafka partitions by the row’s [primary key]({% link {{ page.version.version }}/primary-key.md %}). To define another key to determine the partition for your messages, use the [`key_column`]({% link {{ page.version.version }}/create-changefeed.md %}#key-column) option.
+    The `updated` option adds an `updated` timestamp to each emitted row. You can also use the [`resolved` option](#resolved-messages) to emit a `resolved` timestamp message to each Kafka partition, or to a separate file at a cloud storage sink. A `resolved` timestamp guarantees that no (previously unseen) rows with a lower update timestamp will be emitted on that partition.
 
-- The `UPDATED` option adds an "updated" timestamp to each emitted row. You can also use the [`resolved` option]({% link {{ page.version.version }}/create-changefeed.md %}#resolved-option) to emit a "resolved" timestamp message to each Kafka partition. A "resolved" timestamp guarantees that no (previously unseen) rows with a lower update timestamp will be emitted on that partition.
+    {{site.data.alerts.callout_info}}
+    Depending on the workload, you can use resolved timestamp notifications on every Kafka partition to provide strong ordering and global consistency guarantees by buffering records in between timestamp closures. Use the `resolved` timestamp to see every row that changed at a certain time.
+    {{site.data.alerts.end}}
 
-    For example:
+#### Define a key column
 
-    ~~~json
-    {"__crdb__": {"updated": "1532377312562986715.0000000000"}, "id": 1, "name": "Petee H"}
-    {"__crdb__": {"updated": "1532377306108205142.0000000000"}, "id": 2, "name": "Carl"}
-    {"__crdb__": {"updated": "1532377358501715562.0000000000"}, "id": 3, "name": "Ernie"}
-    {"__crdb__":{"resolved":"1532379887442299001.0000000000"}}
-    {"__crdb__":{"resolved":"1532379888444290910.0000000000"}}
-    {"__crdb__":{"resolved":"1532379889448662988.0000000000"}}
-    ...
-    {"__crdb__":{"resolved":"1532379922512859361.0000000000"}}
-    {"__crdb__": {"updated": "1532379923319195777.0000000000"}, "id": 4, "name": "Lucky"}
-    ~~~
+Typically, changefeeds that emit to Kafka sinks shard rows between Kafka partition using the row's primary key. The primary key is computed into a hash, which remains the same and ensures a row will always emit to the same Kafka partition.
```

Comment:
> computed into a hash

nit: "hashed"

Reply:
Done! I always shy away from making tech terms verbs, because of translation, but this is common enough that it should not be a problem.